### PR TITLE
Add context to dependencies endpoint

### PR DESCRIPTION
**Pull request by yoave23**
## Which problem is this PR solving?
- resolves https://github.com/jaegertracing/jaeger/issues/2369 - auth token is not propagated in `/dependencies` endpoint

## Short description of the changes
- add a `context.Context` param to the `GetDependencies` interface and implement accordingly

### CODE REVIEW

File: cmd/query/app/ui/placeholder/gen_assets.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ var _escData = map[string]*_escFile{
 		name:    "index.html",
```
Comment (yurishkuro) [inline cmd/query/app/ui/placeholder/gen_assets.go:213]: please exclude this file from the PR, it's not related

File: plugin/storage/es/dependencystore/storage.go [hunk 2]
```diff
@@ -81,13 +81,13 @@ func (s *DependencyStore) writeDependencies(indexName string, ts time.Time, depe
...
 // GetDependencies returns all interservice dependencies
-func (s *DependencyStore) GetDependencies(endTs time.Time, lookback time.Duration) ([]model.DependencyLink, error) {
+func (s *DependencyStore) GetDependencies(ctx context.Context, endTs time.Time, lookback time.Duration) ([]model.DependencyLink, error) {
 	indices := getIndices(s.indexPrefix, endTs, lookback)
 	searchResult, err := s.client.Search(indices...).
 		Size(10000). // the default elasticsearch allowed limit
 		Query(buildTSQuery(endTs, lookback)).
 		IgnoreUnavailable(true).
-		Do(s.ctx)
+		Do(ctx)
```
Comment (yurishkuro) [inline plugin/storage/es/dependencystore/storage.go:90]: Good catch. Can we delete the field `s.ctx` altogether? I think it was implemented as anti-pattern.

File: plugin/storage/es/mappings/gen_assets.go [hunk 3]
```diff
@@ -213,7 +213,7 @@ var _escData = map[string]*_escFile{
 		name:    ".nocover",
```
Comment (yurishkuro) [inline plugin/storage/es/mappings/gen_assets.go:213]: please exclude this file

File: plugin/storage/es/dependencystore/storage.go (hunk 2 above)
Comment (yoave23) [reply plugin/storage/es/dependencystore/storage.go:88]: deleted

**Pull request merged**